src/bokeh/util/info.py

```diff
@@ -57,6 +60,33 @@ def print_info() -> None:
     print(f"jupyter_bokeh version :  {_if_installed(_version('jupyter_bokeh', '__version__'))}")
     print(f"Operating system      :  {platform.platform()}")
 
+def print_non_default_settings() -> None:
+    """ Print non-default settings in a table format. """
+
+    all_settings = get_all_settings()
+
+    non_default_settings = []
+    for name, descriptor in all_settings.items():
+        try:
+            if descriptor() != descriptor._convert(cast(str, descriptor.default)):
```

Comment:
OK I think the `cast` points to needing a proper API for determining whether a setting value has been user-set or not. I'd propose trying adding this to the setting class:
```python
    @property
    def is_set(self) -> bool:
        return self._user_value is _Unset
```
Then this condition here can just be 
```python
if not descriptor.is_set:
    non_default_settings.append((name, descriptor()))
```
It would be good to add a trivial test to `test_settings.py` for this new property (or possibly just a couple of `assert`s to existing tests)

Once this last change is done I think this PR will be ready, thanks for the patient iteration @gkarabelos